src/packages/core/src/open-telemetry/entity.ts

```diff
@@ -0,0 +1,65 @@
+import { Entity, Field } from '../decorators';
+import { ID, Sort } from '../types';
+
+import {
+	GraphQLJSON,
+	GraphQLNanoTimestamp,
+	GraphQLNanoDuration,
+} from '@exogee/graphweaver-scalars';
+
+export const addTraceEntityToSchema = () => {
+	// We are applying the decorators to the Trace class here if tracing is enabled.
+	Field(() => ID)(Trace.prototype, 'id');
+	Field(() => String, { adminUIOptions: { hideInFilterBar: true, hideInTable: true } })(
+		Trace.prototype,
+		'spanId'
+	);
+	Field(() => String, {
+		nullable: true,
+		adminUIOptions: { hideInFilterBar: true, hideInTable: true },
+	})(Trace.prototype, 'parentId');
+	Field(() => String)(Trace.prototype, 'name');
+	Field(() => GraphQLNanoTimestamp, { adminUIOptions: { hideInFilterBar: true } })(
+		Trace.prototype,
+		'timestamp'
+	);
+	Field(() => GraphQLNanoDuration, { adminUIOptions: { hideInFilterBar: true } })(
+		Trace.prototype,
+		'duration'
+	);
+	Field(() => String)(Trace.prototype, 'traceId');
+	Field(() => GraphQLJSON, { adminUIOptions: { hideInFilterBar: true, hideInTable: true } })(
+		Trace.prototype,
+		'attributes'
+	);
```

Comment:
Could we just factory the whole entity here so our decorators look normal, or no?

Reply:
Yeah the reason for doing it this way is that the the developer can just apply ACL to the class as normal:

```
import { Trace } from '@exogee/graphweaver';
ApplyAccessControlList({
	DARK_SIDE: {
		// Dark side user role can perform operations on any tag
		all: true,
	},
})(Trace);
```

So it looks odd in this file but better from the end user.